File: src/main/java/com/example/demo/model/Categories.java
package com.example.demo.model;

import java.io.Serializable;
import java.util.List;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Setter
public class Categories implements Serializable {
    private static final long serialVersionUID = 1L;
    @Getter
    private List<Category> items;

    public Categories(List<Category> arg) {
        this.items = arg;
    }
}

File: src/const/categories.ts
export const CATEGORIES = [
  "食費",
  "日用",
  "趣味",
  "交通",
  "交際",
  "医療",
  "教育",
  "通信",
  "水道光熱",
  "住居",
  "被覆",
  "保険",
  "特別",
  "雑費",
  "未設定"
] as const;

export type Category = typeof CATEGORIES[number];
